index.js

```diff
@@ -102,6 +110,59 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  if (!fastify.hasRequestDecorator('pg')) {
+    fastify.decorateRequest('pg', null)
+  }
+
+  fastify.addHook('onRoute', routeOptions => {
+    const transactionConnection = routeOptions.pg && routeOptions.pg.transact
+
+    if (transactionConnection) {
+      const preHandler = async (req, reply) => {
+        const client = await pool.connect()
+
+        if (name) {
```

Comment:
are you sure about the namespacing logic? you must handle all combinations of:

- non namespaced / namespaced registration
- non namespaced / namespaced route transaction

to make an example of something that doesn't work correctly here, if you register with name `foo` and enable the transaction with name `bar`, this code should do nothing.

Reply:
Good spot.

An error might be good in this case too, but I should also make a test that covers it.

Currently working through the test for duplicate namespaces that might have exposed something else, so I will work on this next.

Reply:
> if you register with name foo and enable the transaction with name bar, this code should do nothing.

I think this can be handled just by checking the `req` for the correct namespace and if it doesn't exist throwing an error.

Either way, I think we should do this, so I will add an error and test for this